### PR TITLE
Fix the dev app server; fix for #559

### LINKED ISSUE #559: The dev app server fails to launch
Fails with this error:

!ENTRY com.google.cloud.tools.eclipse.appengine.localserver 4 1 2016-08-15 11:35:31.183
!MESSAGE Google Cloud SDK: Error terminating server: java.net.ConnectException: Connection refused

### PR DESCRIPTION
**Pull request by nbashirbello**
Updating the plugin.xml to fix some merge errors in earlier PR: Removed the profile mode; publish task extension point was pointing to the wrong class (a class that was renamed)

close #559

### CODE REVIEW

**Comment by nbashirbello** [comment]
@briandealwis @elharo

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml [hunk 1]
```diff
@@ -104,7 +104,7 @@
    <extension
          point="org.eclipse.wst.server.core.publishTasks">
       <publishTask
-            class="com.google.cloud.tools.eclipse.appengine.localserver.server.CloudSdkPublishTaskDelegate"
+            class="com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishTaskDelegate"
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml]: So is the problem actually the stubs or is it this? Or is it both of them together?

Comment (nbashirbello) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml]: There were 2 errors causing the dev app server failing: (i) the project wasn't getting published (ii) after being published there were com/google/appengine/tools/development/DevSocketImplFactory error. This PR fixes both.

**Comment by elharo** [comment]
LGTM

**Pull request merged**